test/test_call.py

```diff
@@ -88,6 +105,258 @@ def g():
+                    return HPy_NULL;
+                }
+
+                HPyTracker_Close(ctx, ht);
+                result = HPyTuple_Pack(ctx, 2, result_0, result_1);
+                HPy_Close(ctx, result_0);
+                HPy_Close(ctx, result_1);
+                return result;
+            }
+            @EXPORT(call)
+            @INIT
+        """)
+
+        test_args = (
+            # (receiver, method, args_tuple)
+            dict(receiver={"hello": 1, "world": 2}, name="keys", args=tuple()),
+            dict(receiver="Hello, World", name="find", args=("Wo", )),
+        )
+
+        for kw in test_args:
+            res = getattr(kw["receiver"], kw["name"])(*kw["args"])
+            assert mod.call(**kw) == (res, res)
+
+        with pytest.raises(AttributeError):
+            mod.call(receiver=dict(), name="asdf", args=tuple())
+
+        with pytest.raises(TypeError):
+            mod.call(receiver="Hello, World", name="find")
+
+        with pytest.raises(TypeError):
+            mod.call(receiver="Hello, World", name="find", args=("1", ) * 100)
+
+    def test_hpy_callvectordict(self):
+        import pytest
+        mod = self.make_module("""
+            #define ARGS_OFFSET 1
+
+            HPyDef_METH(call, "call", call_impl, HPyFunc_KEYWORDS)
+            static HPy call_impl(HPyContext *ctx, HPy self,
+                                 HPy *args, HPy_ssize_t nargs, HPy kw)
+            {
+                HPyTracker ht;
+                HPy result;
+                HPy m_kw = HPy_NULL;
+                HPy_ssize_t m_nargs = nargs - ARGS_OFFSET;
+                HPy *m_args = NULL;
+
+                if (nargs < ARGS_OFFSET) {
+                    HPyErr_SetString(ctx, ctx->h_TypeError, "HPy_CallVectorDict requires a receiver");
+                    return HPy_NULL;
+                }
+                m_args = args + ARGS_OFFSET;
+
+                static const char *kwlist[] = { "kw", NULL };
+                if (!HPyArg_ParseKeywords(ctx, &ht, NULL, 0, kw, "|O", kwlist, &m_kw)) {
+                    return HPy_NULL;
+                }
+                result = HPy_CallVectorDict(ctx, args[0], m_args, m_nargs, m_kw);
+                HPyTracker_Close(ctx, ht);
+                return result;
+            }
+            @EXPORT(call)
+            @INIT
+        """)
+
+        def foo():
+            raise ValueError
+
+        def listify(*args):
+            return args
+
+        def f(a, b):
+            return a + b
+
+        def g():
+            return "this is g"
+
+        class KwDict(dict):
+            def __getitem__(self, key):
+                return "key=" + str(key);
+
+        test_args = (
+            # (receiver, method, args_tuple)
+            (dict, (dict(a=0, b=1), ), {}),
+            (dict, tuple(), dict(a=0, b=1)),
+        )
+        for receiver, args_tuple, kw in test_args:
+            assert mod.call(receiver, *args_tuple, kw=kw) == receiver(*args_tuple, **kw)
+
```

Comment:
maybe it's just me, but I find this style of testing very difficult to read. To start with the comment `# (receiver, method, args_tuple)` looks wrong.
I think the following is orders or magnitude easier to read.
```python
assert mod.call(dict, dict(a=0, b=1), kw={}) == dict((dict(a=0, b=1)) # or even == {'a': 0, 'b': 1}
assert mod.call(dict, kw=dict(a=0, b=1)) == dict(a=0, b=1)
```
If the code above is not equivalent to the original code, it's a good sign that the original code is too complex to read 😅.

Feel free to ignore this suggestion though.